crates/bevy_asset/src/ser.rs

```diff
@@ -0,0 +1,87 @@
+use std::cell::Cell;
+
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::{Asset, AssetServer, Handle};
+
+///////////////////////////////////////////////////////////////////////////////
+
+thread_local! {
+    static ASSET_SERVER: Cell<Option<AssetServer>> = Cell::new(None);
+}
+
+impl<T: Asset> Serialize for Handle<T> {
```

Comment:
Maybe later, I do think it's complexity enough for now

Reply:
If you're supporting `Handle` serialization, you may as well support `HandleUntyped` serialization as well. 
Having one without the other is an asymmetry I'd like to avoid.